diff --git a/pip/_internal/utils/setuptools_build.py b/pip/_internal/utils/setuptools_build.py
index 1b8c7a1c2..57c6dde32 100644
--- a/pip/_internal/utils/setuptools_build.py
+++ b/pip/_internal/utils/setuptools_build.py
@@ -104,36 +104,6 @@ def make_setuptools_clean_args(
     return args
 
 
-def make_setuptools_develop_args(
-    setup_py_path: str,
-    *,
-    global_options: Sequence[str],
-    no_user_config: bool,
-    prefix: str | None,
-    home: str | None,
-    use_user_site: bool,
-) -> list[str]:
-    assert not (use_user_site and prefix)
-
-    args = make_setuptools_shim_args(
-        setup_py_path,
-        global_options=global_options,
-        no_user_config=no_user_config,
-    )
-
-    args += ["develop", "--no-deps"]
-
-    if prefix:
-        args += ["--prefix", prefix]
-    if home is not None:
-        args += ["--install-dir", home]
-
-    if use_user_site:
-        args += ["--user", "--prefix="]
-
-    return args
-
-
 def make_setuptools_egg_info_args(
     setup_py_path: str,
     egg_info_dir: str | None,
