diff --git a/pip/_internal/network/download.py b/pip/_internal/network/download.py
index c158650a6..521abb76a 100644
--- a/pip/_internal/network/download.py
+++ b/pip/_internal/network/download.py
@@ -18,7 +18,14 @@ from pip._vendor.urllib3._collections import HTTPHeaderDict
 from pip._vendor.urllib3.exceptions import ProtocolError, ReadTimeoutError
 
 from pip._internal.cli.progress_bars import BarType, get_download_progress_renderer
-from pip._internal.exceptions import IncompleteDownloadError, NetworkConnectionError
+from pip._internal.exceptions import (
+    ConnectionFailedError,
+    ConnectionTimeoutError,
+    IncompleteDownloadError,
+    NetworkConnectionError,
+    ProxyConnectionError,
+    SSLVerificationError,
+)
 from pip._internal.models.link import (
     Link,
     PathComponent,
@@ -266,7 +273,24 @@ class Downloader:
                         raise IncompleteDownloadError(download)
 
                 self._process_response(download, resume_resp)
-            except (ConnectionError, ReadTimeoutError, ProtocolError, OSError):
+            except (
+                ConnectionFailedError,
+                ConnectionTimeoutError,
+                ProxyConnectionError,
+                SSLVerificationError,
+                ReadTimeoutError,
+                ProtocolError,
+                OSError,
+            ):
+                # The error handling here is tricky, a few notes:
+                #
+                # - The diagnostic connection errors are raised by our custom
+                #   requests.request() connection exception handler.
+                # - ProtocolError is raised by urllib3 when the returned data length
+                #   doesn't match the response Content-Length.
+                # - ReadTimeoutError / ProtocolError come straight from urllib3 (via
+                #   process_response) and aren't caught by our connection exception
+                #   handler since they occur while *streaming* a response.
                 continue
 
         # No more resume attempts. Raise an error if the download is still incomplete.
