diff --git a/pip/_internal/req/req_install.py b/pip/_internal/req/req_install.py
index c92c5cff0..5e94679a6 100644
--- a/pip/_internal/req/req_install.py
+++ b/pip/_internal/req/req_install.py
@@ -179,6 +179,8 @@ class InstallRequirement:
         self.requirements_to_check: list[str] = []
 
         # The PEP 517 backend we should use to build the project
+        self._pep517_backend_spec: str
+        self._pep517_backend_path: str | None
         self.pep517_backend: BuildBackendHookCaller | None = None
 
         # This requirement needs more preparation before it can be built
@@ -495,11 +497,21 @@ class InstallRequirement:
         requires, backend, check, backend_path = pyproject_toml_data
         self.requirements_to_check = check
         self.pyproject_requires = requires
+        self._pep517_backend_spec = backend
+        self._pep517_backend_path = backend_path
+
+    def configure_backend(self, python_executable: str) -> None:
+        """Set up the build backend hook caller.
+
+        This is done separately after pyproject.toml loading as the backend
+        need to be called with the build environment's Python executable,
+        which can vary."""
         self.pep517_backend = ConfiguredBuildBackendHookCaller(
             self,
             self.unpacked_source_directory,
-            backend,
-            backend_path=backend_path,
+            self._pep517_backend_spec,
+            backend_path=self._pep517_backend_path,
+            python_executable=python_executable,
         )
 
     def editable_sanity_check(self) -> None:
@@ -773,6 +785,7 @@ class InstallRequirement:
         warn_script_location: bool = True,
         use_user_site: bool = False,
         pycompile: bool = True,
+        script_executable: str | None = None,
     ) -> None:
         # Lazy import to avoid transitively importing `_vendor.distlib.compat`
         # which in turn imports `urllib.request` which is slow.
@@ -803,6 +816,7 @@ class InstallRequirement:
             warn_script_location=warn_script_location,
             direct_url=self.download_info if self.is_direct else None,
             requested=self.user_supplied,
+            script_executable=script_executable,
         )
         self.install_succeeded = True
 
