diff --git a/pip/_internal/commands/index.py b/pip/_internal/commands/index.py
index ce0bafda3..e444b9a74 100644
--- a/pip/_internal/commands/index.py
+++ b/pip/_internal/commands/index.py
@@ -16,7 +16,12 @@ from pip._internal.commands.search import (
     get_installed_distribution,
     print_dist_installation_info,
 )
-from pip._internal.exceptions import CommandError, DistributionNotFound, PipError
+from pip._internal.exceptions import (
+    CommandError,
+    DiagnosticPipError,
+    DistributionNotFound,
+    PipError,
+)
 from pip._internal.index.collector import LinkCollector
 from pip._internal.index.package_finder import PackageFinder
 from pip._internal.models.selection_prefs import SelectionPreferences
@@ -80,6 +85,8 @@ class IndexCommand(IndexGroupCommand):
         # Error handling happens here, not in the action-handlers.
         try:
             handler_map[action](options, args[1:])
+        except DiagnosticPipError:
+            raise
         except PipError as e:
             logger.error(e.args[0])
             return ERROR
